Interface/SamuraiHudWindow.cs

```diff
@@ -0,0 +1,155 @@
+using Dalamud.Game.ClientState.Structs.JobGauge;
+using Dalamud.Plugin;
+using ImGuiNET;
+
+namespace DelvUIPlugin.Interface {
+    public class SamuraiHudWindow : HudWindow {
+        public override uint JobId => 38;
+        
+        private new static int BarHeight => 26;
+        private new static int BarWidth => 357;
+        private new static int XOffset => 178;
+        private new static int YOffset => 496;
+        
+        public SamuraiHudWindow(DalamudPluginInterface pluginInterface, PluginConfiguration pluginConfiguration) : base(pluginInterface, pluginConfiguration) { }
+
+        protected override void Draw(bool _) {
+            DrawHealthBar();
+            DrawPrimaryResourceBar();
+            DrawSenResourceBar();
+            DrawMeditationResourceBar();
+            DrawTargetBar();
+        }
+
+        protected override void DrawPrimaryResourceBar()
+        {
+            var gauge = PluginInterface.ClientState.JobGauges.Get<SAMGauge>();
+
+            // const int xPadding = 5;
+            const int yPadding = 3;
+            var barWidth = BarWidth;
+            var xPos = CenterX - XOffset;
+            var yPos = CenterY + YOffset - BarHeight - yPadding;
+            var cursorPos = new Vector2(xPos, yPos);
+            const int chunkSize = 50;
+            var barSize = new Vector2(barWidth, BarHeight);
+
+            // Chunk 1
+            var Kenki = Math.Min((int)gauge.Kenki, chunkSize);
+            var scale = (float)Kenki / chunkSize;
+            var drawList = ImGui.GetWindowDrawList();
+            drawList.AddRectFilled(cursorPos, cursorPos + barSize, 0x88000000);
+
+            if (scale >= 1.0f)
+            {
+                drawList.AddRectFilledMultiColor(
+                    cursorPos, cursorPos + new Vector2(barWidth * scale, BarHeight),
+                    0xFF5252FF, 0xFF9C9CFF, 0xFF9C9CFF, 0xFF5252FF
+                );
+            }
+            else
+            {
+                drawList.AddRectFilledMultiColor(
+                    cursorPos, cursorPos + new Vector2(barWidth * scale, BarHeight),
+                    0xFF90827C, 0xFF8E8D8F, 0xFF8E8D8F, 0xFF90827C
+                );
+            }
+
+            drawList.AddRect(cursorPos, cursorPos + barSize, 0xFF000000);
+
+        }
+
+        private void DrawSenResourceBar() {
+            var gauge = PluginInterface.ClientState.JobGauges.Get<SAMGauge>();
+
+            const int xPadding = 3;
+            const int yPadding = 3;
+            const int numChunks = 3;
+            
+            var barWidth = (BarWidth - xPadding * (numChunks - 1)) / numChunks;
+            var barSize = new Vector2(barWidth, BarHeight);
+            var xPos = CenterX - XOffset;
+            var yPos = CenterY + YOffset ;
+            var cursorPos = new Vector2(xPos - xPadding - barWidth, yPos);
+            
+            var drawList = ImGui.GetWindowDrawList();
+            
+            for (var i = 1; i < 4; i++) {
```

Comment:
Why are you looping over this? I don't think we're gaining anything in clarity or brevity here. 

Personally I think we remove the loop and the `if i == #` checks and just have the duplicated cursorpos and border rect lines.